salesforce.go

```diff
@@ -161,16 +172,31 @@ func validateBulk(sf Salesforce, records any, batchSize int, isFile bool) error
 	return nil
 }
 
-func processSalesforceError(resp http.Response) error {
-	var errorMessage string
+func processSalesforceError(resp http.Response, auth *authentication, payload requestPayload) (*http.Response, error) {
```

Comment:
TODO: run some tests to make sure this is still capturing errors as expected

It should capture errors both on the first try, and after a refreshed session